i18next-scanner.config.cjs

```diff
@@ -0,0 +1,77 @@
+/* eslint-disable @typescript-eslint/no-var-requires */
+const path = require('path');
+const fs = require('fs');
+const typescript = require('typescript');
+
+module.exports = {
+  input: [
+    'src/**/*.{ts,tsx}',
+    // Use ! to filter out files or directories
+    '!**/node_modules/**',
+  ],
+  output: './',
+  options: {
+    debug: true,
+    createOldCatalogs: false,
+    removeUnusedKeys: true,
+    func: {
+      list: ['i18next.t', 'i18n.t', 't'],
+      extensions: ['.js', '.jsx'], // don't pass ts or tsx here!
+    },
+    trans: {
+      component: 'Trans',
+      i18nKey: 'i18nKey',
+
+      extensions: ['.jsx'], // don't pass ts or tsx here!
+    },
+    lngs: ['en'],
+    ns: ['common'],
+    defaultLng: 'en',
+    defaultNs: 'common',
+    defaultValue: '',
+    resource: {
+      loadPath: 'src/core/localization/translations/{{lng}}/{{ns}}.json',
+      savePath: 'src/core/localization/translations/{{lng}}/{{ns}}.json',
+      jsonIndent: 2,
+      lineEnding: '\n',
+    },
+    pluralSeparator: ':',
+    contextSeparator: ':',
+    nsSeparator: ':',
+    keySeparator: '.',
+    interpolation: {
+      prefix: '{{',
+      suffix: '}}',
+    },
+  },
+
+  transform: typescriptTransform(),
+};
+
+function typescriptTransform(
+  options = {
+    tsOptions: {
+      target: 'es2018',
```

Comment:
yeah, it would be better
but in gives me next error 
```
i18next-scanner: Unable to parse Trans component:
import { createAtom } from '~utils/atoms/createPrimitives';
import { currentMapAtom } from '~core/shared_state';
import { setMapInteractivity } from '~utils/map/setMapInteractivity';
import { registerMapListener } from '~core/shared_state/mapListeners';
let isAtomEnabled = false;
export const clickCoordinatesAtom = createAtom({
    currentMapAtom,
    _set: (coords) => coords,
    start: () => null,
    stop: () => null,
}, ({ onChange, onAction, schedule, create, get }, state = null) => {
    const enableListeners = (map) => schedule((dispatch, ctx = {}) => {
        ctx.removeClickListener ??= registerMapListener('click', (e) => {
            dispatch(create('_set', {
                lng: e.lngLat.lng,
                lat: e.lngLat.lat,
            }));
            return false;
        }, 10);
        setMapInteractivity(map, false);
    });
    const disableListeners = (map) => schedule((dispatch, ctx = {}) => {
        setMapInteractivity(map, true);
        ctx.removeClickListener?.();
        delete ctx.removeClickListener;
    });
    onAction('_set', (coords) => {
        state = coords;
    });
    onAction('start', () => {
        isAtomEnabled = true;
        const map = get('currentMapAtom');
        if (!map)
            return;
        enableListeners(map);
    });
    onAction('stop', () => {
        isAtomEnabled = false;
        const map = get('currentMapAtom');
        state = null;
        if (!map)
            return;
        disableListeners(map);
    });
    onChange('currentMapAtom', (map) => {
        if (!map)
            return;
        if (isAtomEnabled) {
            enableListeners(map);
        }
    });
    return state;
});
```
It looks like it can't parse `??=`  if "target": "esnext"

Reply:
hey, I've found of how to reuse tsconfig + avoid error. It was just acorn version too low to support latest syntax. Now works cool.